source/AI.cpp

```diff
@@ -3142,6 +3220,28 @@ double AI::RendezvousTime(const Point &p, const Point &v, double vp)
 }
 
 
+// Searches every asteroid within the ship scan limit and returns the asteroid
+// closest to the ship.
+bool AI::TargetAsteroid(Ship &ship) const
+{
+	double scanLimit = 100. * sqrt(ship.Attributes().Get("asteroid scan power"));
+	if(!scanLimit)
+	{
+		return false;
+	}
+	for(const shared_ptr<Minable> &asteroid : minables)
+	{
+		double range = ship.Position().Distance(asteroid->Position());
+		if(range < scanLimit)
+		{
+			ship.SetTargetAsteroid(asteroid);
+			scanLimit = range;
+		}
+	}
+	return ship.GetTargetAsteroid().get();
```

Comment:
I get an error when I remove `.get()` that it can't be cast to bool.  I'm not sure I understand by what you mean by "unboxing".  Can you elaborate on the term?